task_registry.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+import multiprocessing as mp
+from dataclasses import dataclass, field, asdict, is_dataclass
+from typing import Any, Callable, Dict, List, Literal, Optional, Protocol, Set, Tuple
+from promise import ScopedConstraint
+
+@dataclass(frozen=True, slots=True)
+class TaskSpec:
+    name: str
+    fn: Callable[..., Any]
```

Comment:
```suggestion
    fn: Callable
```
prefer no args instead of ellipses